src/partition/InputLines.cc

```diff
@@ -92,15 +93,23 @@ struct Block {
 
     Block() : file(), offset(0), size(0), head(), tail() {}
 
-    CharPtrPair const read(char *buf, bool skipFirstLine);
+    CharPtrPair const read(char *buf, bool skipFirstLine, std::vector<std::string> paramNames = {});
```

Comment:
(for efficiency) Redeclare the last parameter as:
```
std::vector<std::string> const& paramNames = std::vector<std::string>()
```
Do not use the `initialization_list`'s constructor `{}` to initialize the vector with the empty content as this might have some (though, minor) performance penalty..